templates/person/show.json.twig

```diff
@@ -0,0 +1,71 @@
+{
+    "@context": "https://schema.org",
+    "@graph": [
+        {
+            "@type": "Person"
+            {% if person.firstname %},"givenName": "{{ person.firstName}}" {% endif %}
+            {% if person.lastName %},"familyName": "{{ person.lastName}}" {% endif %}
+            ,"gender": "{{ person.gender }}"
+            {% if (person.wikipediaUrl) or (person.viafUrl) %}
+            ,"sameAs": [
+                {% if person.wikipediaUrl %} "{{ person.wikipediaUrl}}"{% endif %}
+                {% if (person.wikipediaUrl) and (person.viafUrl) %} , {% endif %}
+                {% if person.viafUrl %} "{{ person.viafUrl}}" {% endif %}
+            ]
+            {% endif %}
+            {% if person.dob %},"birthDate": "{{person.dob}}" {% endif %}
+            {% if person.dod %}, "deathDate": "{{person.dod}}" {% endif %}
+            {% if person.cityOfBirth %}, "birthPlace": "{{person.cityOfBirth}}" {% endif %}
+            {% if person.cityOfDeath %}, "deathPlace": "{{person.cityOfDeath}}" {% endif %}
+            ,"@id": "{{ url('person_show', {'id':person.id}) }}"
+        }
+        {% for titleRole in person.getTitleRoles("Author") %}
+         {% if titleRole.getRole.getName == "Author" %}
+                {% set book = titleRole.title %}
+                {% if book %}
+                ,{
+                    "@type": "Book",
+                    "@id": "{{ url('title_show', {'id':book.id}) }}" ,
+                    "name": "{{book.title}}"
+                    {% if book.genre %} ,"genre": "{{book.genre}}" {% endif %}
+                    {% if book.pubdate %},"datePublished": "{{book.pubdate}}" {% endif %},
+                    "locationCreated": {
+                        "@type": "Place"
+                        {% if book.locationOfPrinting %}, "name": "{{book.locationOfPrinting}}" {% endif %},
+                        "geo": {
+                            "@type": "GeoCoordinates",
+                            "latitude": "{{book.locationOfPrinting.latitude}}",
+                            "longitude": "{{book.locationOfPrinting.longitude}}"
+                        }
+                    }
+                     ,"author": [
+                        {% for titleRole in book.getTitleRoles("Author") %}{% set author = titleRole.person %}
+                        {
```

Comment:
I'm not sure this is valid JSON-LD. From what I can tell, the author should always have a Person as a value, so you may need to include the minimal amount of data:

```twig
"author": [
{
"@type": "Person",
"@id": {{ url('person_show', {'id': author.id}) }}
}
]
```